spec/javascripts/govuk_publishing_components/analytics-ga4/ga4-form-change-tracker.spec.js

```diff
@@ -0,0 +1,252 @@
+          action: 'select'
+        },
+        'event_data'
+      )
+    })
+
+    it('when a checkbox is deselected', () => {
+      createFormAndSetup(container, 'checkbox')
+
+      const index = {
+        index_section: 0,
+        index_section_count: 1
+      }
+
+      const checkbox = container.querySelector('input[type="checkbox"]')
+      checkbox.setAttribute('data-ga4-index', JSON.stringify(index))
+
+      form.triggerChange('input[type="checkbox"]')
+
+      mockGa4SendData.calls.reset()
+
+      form.triggerChange('input[type="checkbox"]')
+
+      expect(mockGa4SendData).toHaveBeenCalledWith(
+        {
+          ...expectedAttributes,
+          ...index,
+          action: 'remove'
+        },
+        'event_data'
+      )
+    })
+
+    describe('when the date component', () => {
+      let inputs
+      const index = {
+        index_section: 0,
+        index_section_count: 1
+      }
+
+      beforeEach(() => {
+        createFormAndSetup(container, 'date')
+
+        inputs = form.querySelectorAll('input')
+
+        inputs.forEach((input) => {
+          input.setAttribute('data-ga4-index', JSON.stringify(index))
+          form.triggerChange(`input[name="${input.name}"]`)
+        })
+      })
+
+      it('is entirely filled in', () => {
+        expect(mockGa4SendData).toHaveBeenCalledWith(
+          {
+            ...expectedAttributes,
+            ...index,
+            action: 'select',
+            text: Array(3).fill(expectedAttributes.text).join('/')
+          },
+          'event_data'
+        )
+      })
+
+      it('is not entirely filled in', () => {
+        mockGa4SendData.calls.reset()
+
+        inputs[0].value = ''
+
+        expect(mockGa4SendData).not.toHaveBeenCalled()
+      })
+    })
+
+    describe('when multiple select', () => {
+      const index = {
+        index_section: 0,
+        index_section_count: 1
+      }
+      let select, options
+
+      beforeEach(() => {
+        createFormAndSetup(container, 'select-multiple')
+
+        select = form.querySelector('select')
+
+        select.setAttribute('data-ga4-index', JSON.stringify(index))
+
+        options = form.querySelectorAll('option')
+        options[0].selected = true
+        options[1].selected = true
+
+        mockGa4SendData.calls.reset()
+      })
+
+      it('has an option deselected', () => {
+        options[1].selected = false
+
+        mockGa4SendData.calls.reset()
+
+        select.dispatchEvent(
+          new CustomEvent('change', {
+            bubbles: true,
+            detail: { value: options[1].value }
+          })
+        )
+
+        expect(mockGa4SendData).toHaveBeenCalledWith(
+          {
+            ...expectedAttributes,
+            ...index,
+            action: 'remove'
+          },
+          'event_data'
+        )
+      })
+    })
+
+    afterEach(() => {
+      container.innerHTML = ''
+    })
+  })
+
+  afterAll(() => {
+    container.remove()
```

Comment:
I think it might be more readable to put the `afterAll` at the start, with the before functions (if possible). Also would it work if the `afterEach` was removed and the `afterAll` changed to `afterEach`?